### PR TITLE
fixes rust-lang#56766

### LINKED ISSUE #56766: Blank document comment in trait raises error different from with that in function.
If I use document comment that doesn't document anything in function,

```Rust
fn test(){
    println!("Hello world");
    ///
}
```

it will raise error:
```
error[E0585]: found a documentation comment that doesn't document anything.
```

But if I use this in trait, the error confuses me.

```Rust
trait User{
    fn test();
    ///
}
fn main() {
    println!("Hello world");
}
```

```
error: expected one of `async`, `const`, `extern`, `fn`, `type`, or `unsafe`, found `}`
 --> src/main.rs:4:1
  |
3 |     ///
  |        - expected one of `async`, `const`, `extern`, `fn`, `type`, or `unsafe` here
4 | }
  | ^ unexpected token
```

### PR DESCRIPTION
**Pull request by saleemjaffer**
fixes #56766

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -6630,6 +6630,22 @@ impl<'a> Parser<'a> {
...
+                    |tok| tok == &token::Token::CloseDelim(token::Brace)) {
+                        let mut err = self.diagnostic().struct_span_err_with_code(
+                            self.span,
+                            "found a documentation comment that doesn't document anything",
+                            DiagnosticId::Error("E0584".into()),
+                        );
+                        err.help("doc comments must come before what they document, maybe a \
+                            comment was intended with `//`?",
+                        );
+                        err.emit();
```
Comment (estebank) [inline src/libsyntax/parse/parser.rs]: For everything between 6636 and 6644 you can replace it with `self.span_fatal_err(self.prev_span, Error::UselessDocComment).emit();`, like in https://github.com/rust-lang/rust/blob/fcfc7963891db5c91035ea527ead2ef998d6781b/src/libsyntax/parse/parser.rs#L5290

**Review by estebank — COMMENTED** [review]
Can you add a test to `src/test/ui/parser/doc-inside-trait-item.rs`:

```
trait User{
    fn test();
    /// empty doc
    //~^ ERROR found a documentation comment that doesn't document anything
}
fn main() {}
```

and run `./x.py test src/test/ui --bless` and add the new test and its corresponding generated `.stderr` file?

File: src/libsyntax/parse/parser.rs (hunk 1 above)
Comment (saleemjaffer) [reply src/libsyntax/parse/parser.rs]: https://github.com/rust-lang/rust/pull/57333#discussion_r249254186

This is the reason I used `struct_span_err_with_code`. Do we not need that now?

**Comment by Dylan-DPC** [comment]
ping from triage @saleemjaffer you need to address the failing test

**Comment by saleemjaffer** [comment]
This was fixed. All tests seem to be passing.

**Comment by Dylan-DPC** [comment]
@estebank this is ready for a review

**Comment by Centril** [comment]
Ping from triage, @estebank, we eagerly await your review!

File: src/libsyntax/parse/parser.rs (hunk 1 above)
Comment (pnkfelix) [reply src/libsyntax/parse/parser.rs]: (indeed, this does not seem like it needs to be a fatal error; I concur with @saleemjaffer's choice to use a non-fatal error here.)

**Comment by estebank** [comment]
Apologies for the unresponsiveness over the past week. I've been sporadically online.

**Pull request merged**